### PR TITLE
First version

### PR DESCRIPTION
**Pull request by fcollot**
This PR provides the first components of the project. It creates a Python package that can be run in GUI or non-GUI mode. In both cases a custom console is provided.
The package also provides a set of unit tests.

If you want to test MusicBox you can do the following:
- Set `MBOX_VIRTUAL_ENV` to TRUE in CMake. This will allow the creation of a virtual Python environment in the build directory in which MusicBox will be installed.
- Build the project.
- Run the tests using 'make test' (or 'ctest --verbose' for more output).
- Run MusicBox from the virtual environment ('Release/venv/bin' on unix, 'Release/venv/scripts' on Windows):
   `./python -m musicbox`
   `./python -m musicbox --no-gui` for console-only version
   `./python -m musicbox --test` as an alternative to running the tests
   `./python -m musicbox --test --no-gui` to test without the GUI (useful for CI servers)

Currently there is only a console but more will come.

### CODE REVIEW

**Review by Florent2305 — CHANGES_REQUESTED** [review]
mbox_install_venv.vcxproj crash in release on VS2022

**Comment by fcollot** [comment]
This was a dependency issue (it was trying to install MusicBox in the virtual env before it was created).
Also while testing on Windows I realized the command line option tests failed because it was expecting `sys.argv` to be mutable (so not a tuple).
Fixed and tested.

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+if(APPLE)
+    cmake_dependent_option(MBOX_MACOS_BUNDLE "Build standalone as a bundle" ON "MBOX_STANDALONE" OFF)
+endif()
+
+cmake_dependent_option(MBOX_VIRTUAL_ENV "Create a virtual environment in the build directory" OFF "NOT MBOX_STANDALONE" OFF)
+
+option(MBOX_USE_QT5 "Use Qt5 instead of Qt6" OFF)
+
+if(MBOX_USE_QT5)
```
Comment (Florent2305) [inline CMakeLists.txt:82]: Not work on windows 11

File: source/package/musicbox/core/config.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        if _pyside_version is None:
+            pyside6_spec = importlib.util.find_spec('PySide6')
+            if pyside6_spec is not None:
+                pyside_version = 6
+            else:
+                pyside2_spec = importlib.util.find_spec('PySide2')
+                if pyside2_spec is not None:
+                    pyside_version = 2
+                else:
+                    raise ModuleNotFoundError("PySide 2 or 6 is required")
```
Comment (Florent2305) [inline source/package/musicbox/core/config.py:34]: It's better to indicate PySide 6 is more recommended than v2

**Review by Florent2305 — APPROVED** [review]
Good Job

File: CMakeLists.txt (hunk 1 above)
Comment (fcollot) [reply CMakeLists.txt:82]: It seems PySide2 cannot be installed on Windows. I'll remove this option for Windows.

**Pull request merged**